internal/permissions/dummy_permission_store.go

```diff
@@ -9,21 +11,36 @@ type dummy struct {
 	permsByNSID map[syntax.NSID]xmaps.Set[syntax.DID]
 }
 
-func (d *dummy) HasPermission(didstr string, nsid string, rkey string, write bool) (bool, error) {
+func (d *dummy) HasPermission(didstr string, nsid string, rkey string, act Action) (bool, error) {
+	if act != Read {
```

Comment:
I think we should get rid of action entirely. I don't think we'll ever support writing to someone else's private data.

Reply:
done!